src/lib/translate.js

```diff
@@ -23,13 +23,14 @@ export default function translate(id, scope, dictionary, fields, plural, fallbac
 	// plural forms:
 	// key: ['plural', 'singular']
 	// key: { none, one, many }
+	// key: { zero, one, other }
 	// key: { singular, plural }
 	if ((plural || plural===0) && value && typeof value==='object') {
 		if (value.splice) {
 			value = value[plural] || value[0];
 		}
 		else if (plural===0 && defined(value.none)) {
```

Comment:
This conditional should take into account whether `value.none` or `value.zero` are defined, similar to the next conditional